### DISCUSSION TITLE
google-cloud-discoveryengine

### DISCUSSION THREAD

**Discussion by Gafterg33** [googleapis/google-cloud-python#12778] [Q&A] [answered]
Hello, 
I have questions about google-cloud-discoveryengine. First of all, can someone advise me, where to find proper documentation about how discoveryengine.converseconversation and summary work? 
Also, i want to add “confidence score” or some kind of score that will show how well the question is matched to returned document
(10 documents). Can you suggest, how to do it in python?

**Comment by parthea** [comment, marked as answer]
The documentation for the `ConverseConversation` is here :

REST
https://cloud.google.com/generative-ai-app-builder/docs/reference/rest/v1/projects.locations.dataStores.conversations/converse#google.cloud.discoveryengine.v1.ConversationalSearchService.ConverseConversation

Client Libraries 
https://cloud.google.com/python/docs/reference/discoveryengine/latest/google.cloud.discoveryengine_v1.services.conversational_search_service.ConversationalSearchServiceClient#google_cloud_discoveryengine_v1_services_conversational_search_service_ConversationalSearchServiceClient_converse_conversation

We have sample code here:
https://github.com/googleapis/google-cloud-python/blob/main/packages/google-cloud-discoveryengine/samples/generated_samples/discoveryengine_v1_generated_conversational_search_service_converse_conversation_sync.py

However, I don't see any tutorials or guides here: https://cloud.google.com/s/results?q=converseconversation&doctype=sample%2Ctutorial&text=%22ConverseConversation%22

Please file a bug in the [public issue tracker for discoveryengine](https://issuetracker.google.com/issues/new?component=911831&template=1480251) to request tutorials or guides for this API.
